src/charm.py

```diff
@@ -1,206 +1,176 @@
+    def _template_files(self):
+        src_dir = Path("src")
+        manifests = [file for file in glob.glob(f"{src_dir}/*.yaml")]
+        return manifests
+
+    @property
+    def _context(self):
+        context = {
+            "app_name": self.model.app.name,
+            "model_name": self.model.name,
+        }
+        return context
+
+    @property
+    def _spark_operator_layer(self) -> Layer:
+        pebble_layer = {
+            "summary": "spark layer",
+            "description": "pebble config layer for spark-k8s",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "Spark Operator layer",
+                    "startup": "enabled",
+                    "command": (
+                        f"/usr/bin/tini -s -- /usr/bin/spark-operator -v=2 "
+                        "-logtostderr "
+                        f"-namespace={self.model.name} "
+                        "-enable-ui-service=true "
+                        "-controller-threads=10 "
+                        "-resync-interval=30 "
+                        "-enable-batch-scheduler=false "
+                        "-enable-metrics=true "
+                        "-metrics-labels=app_type "
+                        f"-metrics-port={self.model.config['metrics-port']} "
+                        "-metrics-endpoint=/metrics "
+                        "-enable-resource-quota-enforcement=false "
+                        "-enable-webhook=true "
+                        f"-webhook-svc-namespace={self.model.name} "
+                        f"-webhook-port={self.model.config['webhook-port']} "
+                        f"-webhook-svc-name={self.model.app.name} "
+                        f"-webhook-config-name={self._mutating_webhook_name} "
+                        f"-webhook-namespace-selector=model.juju.is/name={self.model.name} "
+                        "-webhook-fail-on-error=true"
+                    ),
                 }
             },
+        }
+        return Layer(pebble_layer)
+
+    def _update_layer(self) -> None:
+        """Updates the Pebble configuration layer if changed."""
+
+        current_layer = self.container.get_plan()
+        new_layer = self._spark_operator_layer
+
+        if current_layer.services != new_layer.services:
+            self.container.add_layer(self._container_name, new_layer, combine=True)
+            try:
+                log.info("Pebble plan updated with new configuration, replanning")
+                self.container.replan()
+            except ChangeError as e:
+                log.error(traceback.format_exc())
+                self.unit.status = BlockedStatus("Failed to replan")
+                raise e
+                return
+
+    def _update_webhook_certs(self) -> None:
+        """Push keys and certs files into spark container"""
+
+        self.container.push("/etc/webhook-certs/ca-cert.pem", self._stored.ca, make_dirs=True)
```

Comment:
I'd add some error handling here in case you are not able to push those files.
Also, I think it is fine if we push files here, but we might also want to consider mounting volumes instead, could be worked in a future PR.